### PR TITLE
Resolve constructor ambiguity in vector()

### PR DESCRIPTION
**Pull request by shreyanshannu18**
Description
This PR addresses the ambiguity in the vector() constructor where passing multiple rings or multiple integers could lead to undefined behavior or silent errors. These changes implement explicit type checking to ensure the user's intent is clear.

Detailed Changes
Ambiguity Guards: Added logic to identify if arg0 and arg1 are both rings or both integers.

Error Handling: Now raises a TypeError if multiple rings are provided or if multiple integers are provided without a data object.

Priority Logic: Maintained standard call formats by prioritizing arg0 as the base ring in vector(ring, object) while allowing fallbacks for vector(object, ring).

Documentation: Added new TESTS cases in the docstring to verify that these specific TypeError exceptions are raised correctly.

Why is this change required?
This change prevents edge cases where a user might accidentally provide two conflicting base rings or degrees, ensuring the vector() constructor remains robust and predictable.

Fixes #6769

### CODE REVIEW

**Comment by shreyanshannu18** [comment]
"Hi everyone, I’m checking in on this PR (#41775). The GitHub CI tests are passing, and I’ve verified the logic against the reported ambiguity. Since the GSoC deadline is nearing, I’d love to get some feedback to ensure this implementation aligns with the project’s standards. Thanks for your time!"

File: src/sage/modules/free_module_element.pyx [hunk 1]
```diff
@@ -198,6 +203,12 @@ def vector(arg0, arg1=None, arg2=None, sparse=None, immutable=False):
         (1, 2, 3)
         sage: v.parent()
         Ambient free module of rank 3 over the principal ideal domain Integer Ring
+
+        Check that the immutable flag is respected for sparse vectors (see :issue:`6769`)::
```
Comment (videlec) [inline src/sage/modules/free_module_element.pyx]: There are two problems with this block: First, the line comment is not indented correctly. Secondly, why is it doing here?

Comment (shreyanshannu18) [reply src/sage/modules/free_module_element.pyx]: "Hi @videlec, thank you for the feedback.

I will correct the indentation of the comment in the next push.

Regarding the placement: I added this check here to verify that the immutable flag is consistent across vector types, but I realize it might be better suited for the sparse vector module. I am re-checking the file structure now to move it to the appropriate location. I'll update the PR shortly!"

**Comment by tscrim** [comment]
I think we should continue the usual GIGO principle: if the user is passing strange things, then that is on the user. Plus in the examples you give, `vector` already errors out with reasonable error messages. So I don't think this extra complexity adds any value.

**Comment by shreyanshannu18** [comment]
Hi @tscrim, thank you for the feedback.
I see your point regarding the GIGO principle. If the current error messages are already considered reasonable for these 'strange' inputs, then I agree that adding extra logic to catch them specifically isn't worth the added complexity.
I will simplify the implementation by removing the 'Ambiguity Guards' and custom TypeErrors. I'll focus this PR on the core fix for the immutable flag in sparse vectors (issue #6769) and the test relocations requested by @videlec. I'll push the updated code shortly!
